changelog/format_detector.go

```diff
@@ -0,0 +1,60 @@
+package changelog
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+)
+
+func DetectPullRequestEntryFormat(changelog string) (string, error) {
+	if _, err := os.Stat(changelog); err == nil {
+		b, err := os.ReadFile(changelog)
+		if err != nil {
+			return "", fmt.Errorf("cannot read file %s", changelog)
+		}
+		changelog = string(b)
+	}
+
+	templates := getPullRequestEntryTemplates()
+
+	for _, template := range templates {
+		if template.pattern.MatchString(changelog) {
+			return template.template, nil
+		}
+	}
+
+	if len(templates) < 1 {
+		return "", fmt.Errorf("cannot detect pull request entry format")
```

Comment:
I think providing an entry with a fallback template is the better option here. Even if it doesn't match the changelog entry format, it would still be faster for the user to edit the format than to enter it manually.

Reply:
I agree! I had the same thought, but couldn't make my mind up on how to handle this.